### ISSUE TITLE
icp cannot copy files larger than 2GB

### ISSUE THREAD

**Issue by iychoi** [irods/irods_client_icommands#315]
Hi, 

I'm using icommands 4.3.0.

When copying a file larger than 2GB using `icp` fails with following error.
```
ubuntu@irodsfs-test:/data$ icp /iplant/home/iychoi/test_2GB.bin /iplant/home/iychoi/tt/test_2GB.bin
remote addresses: 206.207.252.46 ERROR: cpUtil: cp error for /iplant/home/iychoi/tt/test_2GB.bin, status = -27000 status = -27000 SYS_COPY_LEN_ERR
```

However, `iput` worked correctly.

The file I tried has 2147483648 bytes which is 1 byte larger than int max (2147483647). So I guess `signed int32` data type was used in offset/size calculation in the `icp` code. This is just my guess.

Thanks,
Illyoung Choi

**Comment by korydraughn** [comment]
You may be correct. We'll add it to the list of things to look at it.

**Comment by trel** [comment]
Initial poking around suggests this works in 4.2.11...

```
$ icp -h | grep Version
iRODS Version 4.2.11                icp

$ iput 3gfile

$ ils -l 3gfile 
  alice             0 demoResc   3221225477 2022-08-13.19:48 & 3gfile

$ icp 3gfile 3gfilecp

$ ils -l 3gfilecp
  alice             0 demoResc   3221225477 2022-08-13.19:48 & 3gfilecp

$ iget 3gfilecp

$ diff 3gfile 3gfilecp

$ echo $?
0
```

**Comment by iychoi** [comment]
I'm using iCommands version 4.2.11 too. 
I guess it's a server-side issue. CyVerse DataStore is using 4.2.8.

$ imiscsvrinfo 
RCAT_ENABLED
relVersion=rods4.2.8
apiVersion=d
rodsZone=iplant

**Comment by trel** [comment]
wait, so 4.2.11 client talking to a 4.2.8 server... sees this error, too?

in addition to your original 4.3.0 client claim at the top of this issue?

**Comment by iychoi** [comment]
Yes, both 4.2.11 client and 4.3.0 client have the same error. That's why I think it's a server-side issue.

**Comment by alanking** [comment]
Okay, that makes sense. 4.2.9 introduced some very dramatic changes to data movement across a number of APIs which almost certainly broke compatibility between the clients and the servers.

We made sure that transfers/requests over federation worked okay between newer and older versions, but we don't make any guarantees about the iCommands working with non-matching server versions.

**Comment by iychoi** [comment]
I tested the copy operation with `gocommands` that implements the iRODS protocol from scratch. It also had the issue. So I think this is not the icommands' compatibility issue.
